TestingApp/Extensions/UIView+Extensions.swift

```diff
@@ -0,0 +1,32 @@
+import UIKit
+
+extension UIView {
+    func hasResponder() -> Bool {
+        hasResponder(self)
+    }
+
+    func hasResponder(_ view: UIView) -> Bool {
+        if view.isFirstResponder {
+            return true
+        }
+        for subview in view.subviews {
+            if hasResponder(subview) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    func superview(by predicate: (UIView) -> Bool) -> UIView? {
+        if let superview {
+            if predicate(superview) {
+                return superview
+            } else {
+                return superview.superview(by: predicate)
+            }
+        }
+
+        return nil
+    }
```

Comment:
let's add some doc about this function and for what purposes it has been created.

Reply:
Addressed 6eb28290fe9844fc6deacbeae58f42bed18df73f.